src/main/webapp/app/therapeuticareas/views/therapeutic.area.questionnaire.view.htm

```diff
@@ -103,119 +37,57 @@
 
                 <div class="question">
                     <div class="input-group">
-                        <input type="text" class="form-control" data-ng-model="newAnswer" placeholder="Enter a new answer..."/>
-                          <span class="input-group-btn">
-                            <button class="btn btn-default" type="button" data-ng-click="addAnswer(questionnaireEntry, newAnswer)"
-                                    data-ng-disabled="!newAnswer" title="Add">
-                                <span class="glyphicon glyphicon-plus"></span>
-                            </button>
-                          </span>
+                        <input id="inp_newAnswer" type="text" class="form-control" data-ng-model="newAnswer"
+                               placeholder="Enter a new answer..."/>
+                              <span class="input-group-btn">
+                                <button id="btn_addAnswer" class="btn btn-default" type="button"
+                                        data-ng-click="addAnswer(questionnaireEntry, newAnswer)"
+                                        data-ng-disabled="!newAnswer" title="Add">
+                                    <span class="glyphicon glyphicon-plus"></span>
+                                </button>
+                              </span>
                     </div>
                 </div>
 
 
-                <div class="answers" ng-if="questionnaireEntry.answers._embeddedItems.length<=4">
-                    <button class="left btn btn-remove" type="button" title="Delete"
-                                data-ng-click="deleteAnswer(questionnaireEntry, answer)"  data-ng-repeat="answer in questionnaireEntry.answers._embeddedItems | orderBy:'id'">
-                        {{answer.answerText}}    <span class="glyphicon glyphicon-remove"></span>
+                <div class="answers" data-ng-if="questionnaireEntry.answers._embeddedItems.length<=4">
+                    <button id="btn_deleteAnswer{{questionnaireEntry.answers._embeddedItems.id}}" class="left btn btn-remove" type="button" title="Delete"
```

Comment:
questionnaireEntry.answers._embeddedItems.id refers to the id on the javascript collection _embeddedItems object as opposed to the answer comprising each collection entry I believe 
pretty sure the answer object for this iteration of the ng-repeat is available at this point. i.e.
ng-attr-id="btn_deleteAnswer_{{answer.sortOrder}}